### PR TITLE
prismlauncher: add withWaylandGLFW override

### PR DESCRIPTION
**Pull request by spikespaz**
The original PR was closed because I messed up the rebase command and accidentally pinged many people after force-pushing.

<https://github.com/NixOS/nixpkgs/pull/253218>

This PR should be complete and ready for merge, it has been rebased on latest `master`.

### CODE REVIEW

**Review by Minion3665 — APPROVED** [review]
I'll carry over my approving review

**Review by getchoo — APPROVED** [review]

**Comment by getchoo** [comment]
Result of `nixpkgs-review pr 259122` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>prismlauncher</li>
    <li>prismlauncher-qt5</li>
  </ul>
</details>

File: pkgs/games/prismlauncher/wrapper.nix
Comment (Scrumplex) [inline pkgs/games/prismlauncher/wrapper.nix:59]: I don't think this assertion can ever fail without manipulating the actual derivation source. Besides, I don't see the appeal of having `shellWrapper` in the first place.

If any future reader wants to understand what `lib.optional shellWrapper makeWrapper` does, it's just gonna lead them to `withWaylandGLFW` anyway. I don't think it helps much with readability, but rather convolutes the derivation with extra assertions like these.

**Review by huantianad — APPROVED** [review]
Looks good to me, looking forward to having this in nixpkgs. Anything blocking this from being merged, @Scrumplex any additional thoughts?

File: pkgs/games/prismlauncher/wrapper.nix
Comment (wegank) [inline pkgs/games/prismlauncher/wrapper.nix:75]:
```suggestion
  env.waylandPreExec = ''
```

To ensure that this appears in the environment with __structuredAttrs enabled.

Comment (spikespaz) [reply pkgs/games/prismlauncher/wrapper.nix:75]: It's this needed specifically for this derivation? I haven't seen `env` used explicitly in many other packages. Could you please elaborate why you think this is important?

**Comment by spikespaz** [comment]
Can someone take another look at this? It's going stale.

**Comment by nyabinary** [comment]
``makeBinaryWrapper``is preferred over ``makeWrapper``

**Comment by getchoo** [comment]
> `makeBinaryWrapper`is preferred over `makeWrapper`

variable exapnsion isn't supported by `makeBinaryWrapper`, which is required here - see [this](https://github.com/NixOS/nixpkgs/issues/172583) issue

**Comment by getchoo** [comment]
> Can someone take another look at this? It's going stale.

i would follow @scrumplex's and wegank's suggestions

File: pkgs/games/prismlauncher/wrapper.nix
Comment (huantianad) [reply pkgs/games/prismlauncher/wrapper.nix:75]: I don't believe this is super needed, but it may be good practice to do. I've been seeing some users including this change in newer code/PRs.

To check my understanding, if `__structuredAttrs` is enabled, then `env.` must be used to export environment variables, but that shouldn't be needed in this case, since `waylandPreExec` is only used as a bash variable, not environment variable right? I cannot find any documentation on this.

**Comment by huantianad** [comment]
I believe Scrumplex's suggestion was already applied, though perhaps you mean removing `shellWrapper` and always adding `makeWrapper`. I think it would be fine to do so, the performance impact is minimal and it would clean up the wrapper.

**Comment by getchoo** [comment]
> I believe Scrumplex's suggestion was already applied, though perhaps you mean removing shellWrapper and always adding makeWrapper

my mistake, that is what i meant. i think this is important to reiterate:

> If any future reader wants to understand what lib.optional shellWrapper makeWrapper does, it's just gonna lead them to withWaylandGLFW anyway. I don't think it helps much with readability, but rather convolutes the derivation with extra assertions like these.

**Review by wegank — APPROVED** [review]

**Comment by spikespaz** [comment]
If there's any more bike-shedding from people who are not in the loop, I will close this.

**Comment by spikespaz** [comment]
This PR is considered finished and I wont be working on it further unless it needs a rebase on latest master. Take it or leave it as it is, I am not seeing any original suggestions that have not been discussed and or remedied already (either in this issue or the prior one).

**Comment by spikespaz** [comment]
> I believe Scrumplex's suggestion was already applied, though perhaps you mean removing `shellWrapper` and always adding `makeWrapper`. I think it would be fine to do so, the performance impact is minimal and it would clean up the wrapper.

This is what I originally had done in the first PR. It was rejected, and therefore I won't be correcting it as this iteration has already been revised under the guidance of the package maintainers.

**Review by Scrumplex — APPROVED** [review]

**Pull request merged**
